src/rpc/blockchain.cpp

```diff
@@ -2130,7 +2136,58 @@ UniValue addkey(const UniValue& params, bool fHelp)
             }
         }
         break;
-    default:
+    case GRC::ContractType::SCRAPER:
+    {
+        std::string status_string = ToLower(params[3].get_str());
+        GRC::ScraperEntryStatus status = GRC::ScraperEntryStatus::UNKNOWN;
+
+        CBitcoinAddress scraper_address;
+        if (!scraper_address.SetString(params[2].get_str())) {
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "Address specified for the scraper is invalid.");
+        }
+
+        if (block_v13_enabled) {
+            CKeyID key_id;
+
+            scraper_address.GetKeyID(key_id);
+
+            if (action == GRC::ContractAction::ADD) {
+                if (status_string == "false") {
+                    status = GRC::ScraperEntryStatus::NOT_AUTHORIZED;
+                } else if (status_string == "true") {
+                    status = GRC::ScraperEntryStatus::AUTHORIZED;
+                } else if (status_string == "explorer") {
+                    status = GRC::ScraperEntryStatus::EXPLORER;
+                } else {
+                    JSONRPCError(RPC_INVALID_PARAMETER, "Status specified for the scraper is invalid.");
+                }
+            } else if (action == GRC::ContractAction::REMOVE) {
+                status = GRC::ScraperEntryStatus::DELETED;
+            }
+
+            contract = GRC::MakeContract<GRC::ScraperEntryPayload>(
+                        action,
+                        uint32_t {2}, // Contract payload version number
+                        key_id,
+                        status);
+
+        } else { // block v13 not enabled
+            if (action == GRC::ContractAction::ADD && !(status_string == "false" || status_string == "true")) {
+                JSONRPCError(RPC_INVALID_PARAMETER, "Status specified for the scraper is invalid.");
+            } else if (action == GRC::ContractAction::REMOVE) {
+                status_string = "false";
+            }
+
+            // This form of ScraperEntryPayload generation matches the payload constructure that uses the Parse
```

Comment:
"constructure" should probably be constructor?

Reply:
Thanks.